assembl/static2/js/app/components/debate/thread/post.jsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Translate } from 'react-redux-i18n';
+import { Row, Col } from 'react-bootstrap';
+import { createSelector } from 'reselect';
+
+import { postSelector } from '../../../selectors';
+import Like from '../../svg/like';
+import Disagree from '../../svg/disagree';
+import DontUnderstand from '../../svg/dontUnderstand';
+import MoreInfo from '../../svg/moreInfo';
+import ProfileLine from '../../common/profileLine';
+import { SHOW_POST_RESPONSES } from '../../../constants';
+
+const postMapStateToProps = createSelector(postSelector, (post) => {
+  return { expanded: post.get('showResponses', SHOW_POST_RESPONSES) };
+});
+
+export const connectPostToState = connect(postMapStateToProps);
+
+export const PostFolded = ({ creator }) => {
+  return <Translate value="debate.thread.foldedPostLink" creatorName={creator.name} />;
+};
+
+export default class Post extends React.Component {
+  render() {
+    const {
+      children,
+      subject,
+      body,
+      parentId,
+      indirectIdeaContentLinks,
+      mySentiment,
+      sentimentCounts,
+      creator,
+      creationDate
+    } = this.props;
+    let count = 1;
+    const totalSentimentsCount =
```

Comment:
Should cache so only calculate once. So inefficient to re-calculate everytime the post renders, which would be a few times.